### PR TITLE
Support centered settings

### PR DESCRIPTION
**Pull request by yoyo837**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style. `windows`
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: components/modal/style/modal.less [hunk 1]
```diff
@@ -133,6 +134,42 @@
...
+.vertical-center-modal {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.vertical-center-modal .ant-modal {
+  top: 0;
+}
+*/
```
Comment (afc163) [inline components/modal/style/modal.less]: 这些代码没用到建议删掉。

File: components/modal/style/modal.less [hunk 2]
```diff
@@ -133,6 +134,42 @@
   }
 }
 
+.vertical-center-modal {
```
Comment (afc163) [inline components/modal/style/modal.less]: 建议改为 `.@{@dialog-prefix-cls}-centered`

File: components/modal/index.zh-CN.md [hunk 3]
```diff
@@ -20,6 +20,7 @@ title: Modal
 | afterClose | Modal 完全关闭后的回调 | function | 无 |
 | bodyStyle | Modal body 样式 | object | {} |
 | cancelText | 取消按钮文字 | string | 取消 |
+| centered | 居中 Modal | Boolean | `false`
```
Comment (afc163) [inline components/modal/index.zh-CN.md]: 垂直居中展示 Modal

File: components/modal/index.en-US.md [hunk 4]
```diff
@@ -62,6 +62,7 @@ The properties of the object are follows:
 | Property | Description | Type | Default |
 | -------- | ----------- | ---- | ------- |
 | cancelText | Text of the Cancel button | string | `Cancel` |
+| centered | Centered Modal | Boolean | `false`
```
Comment (afc163) [inline components/modal/index.en-US.md]: 加的位置和中文文档不一致。

File: components/modal/style/modal.less [hunk 5]
```diff
@@ -133,6 +134,42 @@
...
+    display: inline-block;
+    vertical-align: middle;
+    top: 0;
+    text-align: left;
+  }
+  .@{table-prefix-cls} {
+    &-body {
+      white-space: nowrap;
+      overflow-x: auto;
+    }
```
Comment (afc163) [inline components/modal/style/modal.less]: Why adding this?

Comment (yoyo837) [reply components/modal/style/modal.less]: The table in modal, `white-space:nowrap;` horizontal scrolling seems to experience better, or do you mean that it is better controlled by the table itself?

**Pull request closed without merging**